### ISSUE TITLE
Feature request: sharable config as a scoped package

### ISSUE THREAD

**Issue by shinnn** [eslint/eslint#2544]
For example, when I (`@shinnn`) publish a sharable config `eslint-config-myconfig` as a [scoped package](https://docs.npmjs.com/misc/scope), npm will place it on `node_modules/@shinnn/eslint-config-myconfig` after installing.

So, if I want to use the config via `extends` option, I should write `.eslintrc` like below:

```json
{
  "extends": "./node_modules/@shinnn/eslint-config-myconfig/index.js"
}
```

I'd like to use a scoped sharable config more simply, just like:

```json
{
  "extends": "@shinnn/myconfig"
}
```

**Comment by nzakas** [comment]
Yeah, this is a bug, we should have had this working.

**Comment by eric-isakson** [comment]
Any reason why we can't have an arbitrary package name for shared configurations? My first thought when I looked at this issue was, hmm...if I just did a require on @shinn/myconfig and the scoped package name was @shinn/myconfig rather than @shinn/eslint-config-myconfig, this wouldn't require any manipulation of the package name used in the extends.

I think the fix for this could either be, manipulate the package name to add eslint-config- to the package name or drop the requirement to prefix configurations with eslint-config- 

To not be a breaking change, for non-scoped packages, we could first check for eslint-config-<package> and if not present fall back to the non-prefixed package.

Thoughts?

**Comment by eric-isakson** [comment]
working on this, should have something committed to my fork tonight

**Comment by eric-isakson** [comment]
I went with a solution that still requires the eslint-config- in the package name. If you are interested in allowing arbitrary package names, I'd be happy to take a look at implementing that, just let me know.

**Comment by analytik** [comment]
Could you please backport this to 3.x?

**Comment by not-an-aardvark** [comment]
This issue was apparently fixed in v0.21.2, so it should already be working in 3.x.
